libzmq/src/error.rs

```diff
@@ -49,34 +40,36 @@ use std::{
 /// ```
 ///
 /// [`ErrorKind`]: enum.ErrorKind.html
-#[derive(Debug)]
-pub struct Error<T = ()> {
-    inner: Context<ErrorKind>,
+#[derive(Debug, Error)]
+#[error("{inner}")]
+pub struct Error<T: Debug = ()> {
```

Comment:
Here adding a manual debug impl would be better so that the `T: Debug` bound is not needed.